interfaces/cython/cantera/test/test_thermo.py

```diff
@@ -1520,3 +1521,156 @@ def test_write_csv(self):
         self.assertEqual(len(data), 7)
         self.assertEqual(len(data.dtype), self.gas.n_species + 2)
         self.assertIn('Y_H2', data.dtype.fields)
+
+        b = ct.SolutionArray(self.gas)
+        b.read_csv(outfile)
+        self.assertTrue(np.allclose(states.T, b.T))
+        self.assertTrue(np.allclose(states.P, b.P))
+        self.assertTrue(np.allclose(states.X, b.X))
+
+    def test_pandas(self):
+        states = ct.SolutionArray(self.gas, 7)
+        states.TPX = np.linspace(300, 1000, 7), 2e5, 'H2:0.5, O2:0.4'
+        try:
+            # this will run through if pandas is installed
+            df = states.to_pandas()
+            self.assertTrue(df.shape[0]==7)
+        except ImportError as err:
+            # pandas is not installed and correct exception is raised
+            pass
+        except Exception as err:
+            raise(err)
+
+    def test_restore(self):
+
+        def check(a, b, atol=None):
+            if atol is None:
+                fcn = lambda c, d: np.allclose(c, d)
+            else:
+                fcn = lambda c, d: np.allclose(c, d, atol=atol)
```

Comment:
It's not identical to `np.allclose`, but in much of the test suite we use a custom function `CanteraTest.assertArrayNear` which provides extra data about failed comparisons, e.g.
```
AssertionError: AssertNear: 354.98584001686 - 999 = -644.01415998314
Relative error for element (3,) of 9.512863e-01 exceeds rtol = 1.000000e-08
```
Rather than the sometimes mysterious `AssertionError: False is not true`.

Reply:
NP - done!